### PR TITLE
layers: Label new pipeline layout consistency VUID

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Adds new VUID for 1.3.244

Note, there was a bug in the Spec and currently being fixed (https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5795) but these are the correct VUID they should be

### CODE REVIEW

File: layers/core_checks/shader_validation.cpp [hunk 1]
```diff
@@ -2699,9 +2718,43 @@ bool CoreChecks::ValidateVariables(const SHADER_MODULE_STATE &module_state) cons
...
                                                   const PIPELINE_STATE &pipeline,
-                                                  const std::vector<ResourceInterfaceVariable> &descriptor_variables,
-                                                  const std::string &vuid_layout_mismatch) const {
+                                                  const std::vector<ResourceInterfaceVariable> &descriptor_variables) const {
     bool skip = false;
+
+    std::string vuid_07988;
+    std::string vuid_07989;
+    std::string vuid_07990;
+    std::string vuid_07991;
```
Comment (ncesario-lunarg) [inline layers/core_checks/shader_validation.cpp:2727]: I wonder if something like the `DrawDispatch*` classes approach would make sense for VUID name lookup depending on how common this scenario is.

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

Heads up @lunarpapillo, as I'm guessing this will effect nightly trace runs when it goes in.

File: layers/core_checks/shader_validation.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/core_checks/shader_validation.cpp:2727]: So I just wanted to get this labeled for the sake of the header update, but once the spec bug gets fixed, I want to just make a `DrawDispatch`-ish class for pipelines, and move these checks all to a single area

**Review by arno-lunarg — APPROVED** [review]
LGTM!

**Comment by mikes-lunarg** [comment]
> Note, there was a bug in the Spec and currently being fixed (https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5795) but these are the correct VUID they should be

Awesome, that explains the missing `VUID-VkRayTracingPipelineCreateInfoKHR-*` VUs I see when running the checker

**Pull request merged**